lib/lightning/projects/provisioner.ex

```diff
@@ -78,6 +80,61 @@ defmodule Lightning.Projects.Provisioner do
     |> maybe_add_project_credentials(user_or_repo_connection)
   end
 
+  defp audit_workflows(project_changeset, user_or_repo_connection) do
+    project_changeset
+    |> get_assoc(:workflows)
+    |> Enum.reduce(
+      Multi.new(),
+      fn workflow_changeset, multi ->
+        additional_multi =
+          classify_audit(workflow_changeset)
+          |> case do
+            {:no_action, _nil} ->
+              Multi.new()
+
+            {action, workflow_id} ->
+              audit_workflow_multi(action, workflow_id, user_or_repo_connection)
+          end
+
+        Multi.append(multi, additional_multi)
```

Comment:
I imagine it would be more readable replacing the case assignment with a function `append_multi` or a smarter `audit_workflow_multi` that does all the logic. Here on the high level would contain only the iteration.

Reply:
Thanks for the feedback @jyeshe - I have pushed the change in a new commit!